src/server/cluster/cluster_slot_migration.cc

```diff
@@ -82,6 +82,31 @@ error_code ClusterSlotMigration::Start(ConnectionContext* cntx) {
   return {};
 }
 
+error_code ClusterSlotMigration::Init(uint32_t sync_id, uint32_t shards_num) {
+  VLOG(1) << "Init slot migration";
+
+  state_ = MigrationState::C_CONNECTING;
+
+  sync_id_ = sync_id;
+  source_shards_num_ = shards_num;
+
+  VLOG(1) << "Resolving host DNS";
+  error_code ec = ResolveHostDns();
```

Comment:
why do we need this both in source node and target node?

Reply:
because I have removed only the first step with config, another communication is still done from the target node to the source node. I will remove it in the future